src/ucp/core/ucp_context.c

```diff
@@ -283,9 +290,15 @@ static ucs_config_field_t ucp_config_table[] = {
    "and the resulting performance.\n",
    ucs_offsetof(ucp_config_t, ctx.estimated_num_ppn), UCS_CONFIG_TYPE_ULUNITS},
 
-  {"RNDV_FRAG_SIZE", "512k",
-   "RNDV fragment size \n",
-   ucs_offsetof(ucp_config_t, ctx.rndv_frag_size), UCS_CONFIG_TYPE_MEMUNITS},
+  UCP_CONFIG_RNDV_FRAG_SIZE(HOST, "512k"),
+  UCP_CONFIG_RNDV_FRAG_SIZE(CUDA, "4M"),
+
+  {"RNDV_FRAG_MEM_TYPE", "host",
+   "RNDV frag memory type.\n"
+   " host      - Use host memory fragments.\n"
+   " cuda      - Use cuda memory fragments.\n"
+   " auto      - Runtime automatically chooses optimal frag mem_type.",
+   ucs_offsetof(ucp_config_t, ctx.rndv_frag_mem_type), UCS_CONFIG_TYPE_ENUM(ucp_rndv_frag_mem_types)},
```

Comment:
maybe define it as a bitmap of which memory types could be used as rndv fragments?
default - all - means use all possible types
then, no need "auto"